src/main/java/org/eclipsefoundation/adopters/service/impl/DefaultAdopterService.java

```diff
@@ -0,0 +1,213 @@
+		Path adoptersFile = Paths.get(adoptersLocation, adoptersName);
+		if (Files.exists(adoptersDir)) {
+			// read in the initial file if it exists
+			if (Files.exists(adoptersFile)) {
+				LOGGER.debug("Found an adopters file at path {}, reading in", adoptersFile);
+				readInFile(adoptersFile);
+			}
+
+			watcher = FileSystems.getDefault().newWatchService();
+			// register the adopter location to watch
+			LOGGER.debug("Registering file watcher on directory {}", adoptersDir);
+			adoptersDir.register(watcher, ENTRY_CREATE, ENTRY_DELETE, ENTRY_MODIFY);
+
+			// start file watching on a new thread
+			new Thread(this::update).start();
+		} else {
+			throw new IllegalArgumentException();
+		}
+	}
+
+	@PreDestroy
+	public void shutdown() {
+		this.running = false;
+	}
+
+	/**
+	 * Update loop, will loop until shutdown is triggered or the service is stopped.
+	 * 
+	 * @throws IOException
+	 */
+	private void update() {
+		// soft lock this to a single thread
+		if (!this.lock.isLocked()) {
+			this.lock.lock();
+			try {
+				// set the state to running at the start
+				this.running = true;
+				// watch for running state to change for gentle shutdowns
+				while (this.running) {
+					WatchKey key = null;
+					try {
+						key = watcher.take();
+						// if there was an error while watching for changes, try again
+						if (key == null) {
+							continue;
+						}
+
+						// pass a filtered list of events to be processed
+						@SuppressWarnings("unchecked")
+						List<WatchEvent<Path>> events = key.pollEvents().stream()
+								.filter(e -> e.kind() != OVERFLOW && e.kind().type() == Path.class)
+								.map(e -> (WatchEvent<Path>) e)
+								.filter(e -> ((Path) e.context()).getFileName().toString().equals(adoptersName))
+								.collect(Collectors.toList());
+						processEvents(events);
+					} catch (InterruptedException e) {
+						// represents shutdown state, so end processing
+						Thread.currentThread().interrupt();
+					} finally {
+						// return the key to the watch service
+						if (key != null && key.isValid()) {
+							key.reset();
+						}
+					}
+				}
+			} finally {
+				// release the lock once leaving the current block
+				this.lock.unlock();
+			}
+		}
+	}
+
+	private void processEvents(List<WatchEvent<Path>> events) {
+		// process each of the queued events
+		for (WatchEvent<Path> event : events) {
+			// clear the adopter list if the event is a deletion event
+			if (event.kind() == ENTRY_DELETE) {
+				LOGGER.debug("Detected the deletion of adopters file, emptying array");
+				synchronized (this) {
+					this.adopters = new ArrayList<>();
+				}
+			} else {
+				// indicates an update or created file on the given
+				readInFile(Paths.get(adoptersLocation, event.context().getFileName().toString()));
+			}
+		}
+	}
+
+	private void readInFile(Path adoptersPath) {
+		LOGGER.debug("Detected an update for adopters file, reading in {}", adoptersPath);
+		Gson gson = new Gson();
```

Comment:
Jsonb isn't included ootb in resteasy, so I could sub out Jackson, but I still need Guava if not GSON for the loading cache since the Quarkus cache is still in its baby stages.